subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/DefaultTypeMetadataStore.java

```diff
@@ -178,15 +178,18 @@ private static String toListOfAnnotations(ImmutableSet<Class<? extends Annotatio
     }
 
     private static class DefaultTypeMetadata implements TypeMetadata {
+        private final Class<?> ownerType;
         private final ImmutableSet<PropertyMetadata> propertiesMetadata;
         private final ReplayingTypeValidationContext validationProblems;
         private final ImmutableMap<Class<? extends Annotation>, ? extends PropertyAnnotationHandler> annotationHandlers;
 
         DefaultTypeMetadata(
+            Class<?> ownerType,
```

Comment:
🧐 Not sure about the "owner" part, what does it mean here? I think using just `type` would be a good solution already.